servers/rendering/renderer_rd/shader_rd.cpp

```diff
@@ -180,6 +180,7 @@ void ShaderRD::_build_variant_code(StringBuilder &builder, uint32_t p_variant, c
 #if defined(MACOS_ENABLED) || defined(IOS_ENABLED)
 				builder.append("#define MOLTENVK_USED\n");
 #endif
+				builder.append(String("#define RENDER_DRIVER_") + OS::get_singleton()->get_current_rendering_driver_name().to_upper() + "\n");
```

Comment:
I think adding the  `RENDER_DRIVER_*` defines here means that shaders that are created via `RenderingDevice::shader_compile_spirv_from_source` alone won't have access to them.

Maybe the defines should be added here: https://github.com/godotengine/godot/blob/eea14a0edc3f2a8ab4a91d1e142d48a5348584bb/modules/glslang/register_types.cpp#L84-L129

Reply:
They do work there. The D3D12 PR (https://github.com/godotengine/godot/pull/64304) already does the same and it works for it.

Reply:
I'm referring to shaders that do not use ShaderRD. For example this shader fails to compile:
```
var rd = RenderingServer.get_rendering_device()
var source = RDShaderSource.new()
source.language = RenderingDevice.SHADER_LANGUAGE_GLSL
source.source_vertex = """
	#version 460

	#ifdef RENDER_DRIVER_VULKAN 
		void main()
		{

		}
	#endif	
"""
var shader_spirv = rd.shader_compile_spirv_from_source(source, false)
print(shader_spirv.compile_error_vertex)
```
ERROR: Linking vertex stage: Missing entry point: Each stage requires one entry point

Reply:
I see.

`RENDER_DRIVER_*` was meant for internal engine use, but it may be worth allowing such an usage. However, in that case, I'd also do the same change to `MOLTENVK_USED` and any other platform or driver dependant macro. My take would be to keep this PR as it is and address the whole macro exposure change in a separate PR.

I'd like to know @clayjohn's thoughts on this.

Reply:
Agreed, let's re-assess this in a follow-up issue/PR.